app/manager-frontend/src/components/ConfigureResultsModifierModal.jsx

```diff
@@ -0,0 +1,161 @@
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  VStack,
+  Alert,
+  AlertIcon,
+  useDisclosure,
+} from "@chakra-ui/react";
+
+import React, { useState } from "react";
+import { Form, Formik, useField } from "formik";
+import { FaArrowRight } from "react-icons/fa";
+import { FormikInput } from "./forms/FormikInput";
+import { FormikSelect } from "./forms/FormikSelect";
+import { LowNumberSuppressionParameters } from "./LowNumberSuppressionParameters";
+import { validationSchema } from "pages/ResultsModifier/validation";
+import { capitaliseObjectKeys } from "helpers/data-structures";
+
+export const ConfigureResultsModifierModal = ({
+  initialData,
+  isOpen,
+  onClose,
+  action,
+}) => {
+  const {
+    isOpen: isConfirmOpen,
+    onOpen: onConfirmOpen,
+    onClose: onConfirmClose,
+  } = useDisclosure();
+  const [feedback, setFeedback] = useState();
+
+  // Todo: Get this from backend
+  const typeOptions = [
+    { id: "Type1", limit: "1" },
+    { id: "Type2", limit: "2" },
+  ];
+
+  const handleSubmit = async (values, actions) => {
+    try {
+      // convert all empty strings to null
+      const payload = Object.entries(values).reduce(
+        (a, [k, v]) => ({
+          ...a,
+          [k]: v !== "" ? v : null,
+        }),
+        {}
+      );
+      // post to the api
+      await action({ values: payload }).json();
+      onClose();
+    } catch (e) {
+      console.error(e);
+      onConfirmClose();
+      setFeedback("Something went wrong!");
+      window.scrollTo(0, 0);
+    }
+    actions.setSubmitting(false);
+  };
+
+  const ConfirmationModal = () => {
```

Comment:
I would even move it outside the component, but keep it in the same file.

like it's a separate component of its own, but in the same file as this is the only place it's used.